### ISSUE TITLE
Fail to put dictionary of tuples

### ISSUE THREAD

**Issue by robertnishihara** [amplab/ray#388]
The following fails.

```python
import ray
ray.init(start_ray_local=True, num_workers=0)
ray.put({(): 1})
```

It crashes on my machine with

```
/Users/rkn/ray/thirdparty/numbuf/python/src/pynumbuf/adapters/python.cc180Check failed: dummy.size() == 0 
```

Note that the following succeeds.

```python
obj = {(): 1}
serialized, _ = ray.serialization.serialize(ray.worker.global_worker.handle, obj)
result = ray.serialization.deserialize(ray.worker.global_worker.handle, serialized)
```

**Comment by robertnishihara** [comment]
Note that this works with the protobuf serialization.

**Comment by robertnishihara** [comment]
This was fixed by #392.
